src/applications/MachineShopSimulator.java

```diff
@@ -154,27 +54,27 @@ static boolean moveToNextMachine(Job theJob) {
     static Job changeState(int theMachine) {// Task on theMachine has finished,
                                             // schedule next one.
         Job lastJob;
-        if (machine[theMachine].activeJob == null) {// in idle or change-over
+        if (machine[theMachine].getActiveJob() == null) {// in idle or change-over
                                                     // state
             lastJob = null;
             // wait over, ready for new job
-            if (machine[theMachine].jobQ.isEmpty()) // no waiting job
+            if (machine[theMachine].getJobQ().isEmpty()) // no waiting job
                 eList.setFinishTime(theMachine, largeTime);
             else {// take job off the queue and work on it
-                machine[theMachine].activeJob = (Job) machine[theMachine].jobQ
-                        .remove();
-                machine[theMachine].totalWait += timeNow
-                        - machine[theMachine].activeJob.arrivalTime;
-                machine[theMachine].numTasks++;
-                int t = machine[theMachine].activeJob.removeNextTask();
+                machine[theMachine].setActiveJob((Job) machine[theMachine].getJobQ()
+                        .remove());
+                machine[theMachine].setTotalWait(machine[theMachine].getTotalWait() + timeNow

```

Comment:
This is a very nasty chain of calls that probably should be cleaned up. I suspect that this indicates that there's logic in the wrong class somewhere.